### PR TITLE
remove dep on statsd client

### PR DESCRIPTION
**Pull request by rf**
Sets up noops instead of using the null reporter which isn't really necessary

r @Raynos

### CODE REVIEW

File: reservoir-backend.js [hunk 1]
```diff
@@ -57,11 +56,16 @@ function ReservoirBackend(options) {
         'options.backend must be object with `logMany` method'
     );
 
-    self.statsd = options.statsd || NullStatsd();
+    self.timing = (options.statsd && options.statsd.timing) || noop;
```
Comment (Raynos) [inline reservoir-backend.js]: this breaks... wrong `this`.

Please write integration test :)

File: reservoir-backend.js [hunk 2]
```diff
@@ -90,12 +86,28 @@ function ReservoirBackend(options) {
         typeof self.timers.setTimeout === 'function',
         'options.timers must be object with setTimeout function'
     );
+    delete options.timers;
```
Comment (Raynos) [inline reservoir-backend.js]: no. don't delete.

File: reservoir-backend.js [hunk 3]
```diff
@@ -103,6 +115,9 @@ function ReservoirBackend(options) {
     self.dropCount = {};
     self.logCount = {};
     self.samplingDecision = null;
+
+    var keys = Object.keys(options);
```
Comment (Raynos) [inline reservoir-backend.js]: don't implement this gaurd. It will create frustration of larch vs debug-larch because they take different options.

Comment (rf) [inline reservoir-backend.js]: Remember, this is the reservoir, not the top level Larch -- there's no such thing as a DebugReservoir. I think it's a pretty cool guard for a public interface.

Comment (Raynos) [inline reservoir-backend.js]: sure; just implement it in a decent away.

aka `subset(Object.keys(options), WHITELIST)`

**Comment by Raynos** [comment]
If I see one more `delete` ...

untitle

**Comment by rf** [comment]
why so serious

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**